cluster-autoscaler/cloudprovider/clusterapi/clusterapi_utils.go

```diff
@@ -264,21 +270,38 @@ func parseIntKey(annotations map[string]string, key string) (resource.Quantity,
 }
 
 func parseCPUCapacity(annotations map[string]string) (resource.Quantity, error) {
-	return parseKey(annotations, cpuKey)
+	// Checking both sets of annotations. If the upstream annotation doesn't exist,
+	// get the value of the deprecated one. If neither exist, return nil.
+	if val, err := parseKey(annotations, cpuKey); val != zeroQuantity && err == nil {
+		return val, err
```

Comment:
If err is nil here, return nil explicitly please
```suggestion
		return val, nil
```